Automattic/wp-calypso
# Store Coupons: fetchCoupons to take params

This changes `fetchCouponsPage` to `fetchCoupons` which can take
parameters other than page so things like `offset` can be used with the
API.

To Test:
1. `npm test`
2. `npm start`
3. Visit http://calypso.localhost:3000/store/promotions/<site url>
4. Inspect Redux dev tools, search for "COUPONS" in actions.
5. Verify actions and that the coupons have been added to state.
